src/transformers/training_args.py

```diff
@@ -615,6 +633,12 @@ def __post_init__(self):
         elif ShardedDDPOption.ZERO_DP_2 in self.sharded_ddp and ShardedDDPOption.ZERO_DP_3 in self.sharded_ddp:
             raise ValueError("`--sharded_ddp zero_dp_2` is not compatible with `--sharded_ddp zero_dp_3`.")
 
+        if self.tpu_metrics_debug:
+            self.debug += " tpu_metrics_debug"
```

Comment:
Raise a deprecation warning here since the argument is documented as deprecated.